src/NUnitFramework/framework/Internal/Filters/CompositeFilter.cs

```diff
@@ -46,16 +47,7 @@ public CompositeFilter()
         /// <param name="filters"></param>
         public CompositeFilter( params TestFilter[] filters )
         {
-            Filters = new List<TestFilter>(filters);
-        }
-
-        /// <summary>
-        /// Adds a filter to the list of filters
-        /// </summary>
-        /// <param name="filter">The filter to be added</param>
-        public void Add(TestFilter filter)
```

Comment:
We are adding logic to filter construction that relies on filters not changing after creation.